kmod/src/data.c

```diff
@@ -695,7 +699,11 @@ static int scoutfs_readpage(struct file *file, struct page *page)
 	ret = scoutfs_lock_inode(sb, SCOUTFS_LOCK_READ, flags, inode,
 				 &inode_lock);
 	if (ret < 0) {
+#ifdef KC_MPAGE_READ_FOLIO
+		unlock_page(&folio->page);
```

Comment:
We should have an on stack 'struct page *page = &folio->page' in the compat case with the folio argument so that most of the page users won't change and we won't need the ifdefs.